source/read-write-concern/tests/README.rst

```diff
@@ -52,6 +52,37 @@ array of test case objects, each of which have the following keys:
 - ``isServerDefault:`` Indicates whether the read or write concern is considered the server's default.
 - ``isAcknowledged:`` Indicates if the write concern should be considered acknowledged.
 
+Aggregation
+~~~~~~~~~~~
+
+These tests are designed to ensure compliance with the spec in relation to what should be
+sent to the server when using the aggregation framework.
+
+Each YAML file contains the following keys:
+
+- ``data``: The data that should exist in the collection under test before each
+  test run.
+- ``minServerVersion`` (optional): The minimum server version (inclusive)
+  required to successfully run the test. If this field is not present, it should
+  be assumed that there is no lower bound on the required server version.
+- ``maxServerVersion`` (optional): The maximum server version (exclusive)
+  against which this test can run successfully. If this field is not present,
+  it should be assumed that there is no upper bound on the required server
+  version.
+- ``tests``: An array of tests that are to be run independently of each other.
+  Each test will have some or all of the following fields:
+  - ``description``: The name of the test.
+  - ``operation``: Document describing the operation to be executed. This will
+    have the following fields:
+    - ``name``: The name of the operation as defined in the specification. The
+      name `db-aggregate` refers to database-level aggregation.
+    - ``arguments``: The names and values of arguments from the specification.
+  - ``error``: If ``true``, the test should expect an error or exception. Note
+      that some drivers may report server-side errors as a write error within a
+      write result object.
+  - ``readConcernDocument:`` A document indicating the read concern to be sent to the server.
```

Comment:
I have a few concerns with these tests:

1. This is adding CRUD tests to the read/write concern spec, which already has their own test format
2. We're introducing a new `readConcernDocument` field, which has no prior art in any other tests and comes off as a a stopgap until we have APM assertions on CRUD tests (see: [SPEC-1133](https://jira.mongodb.org/browse/SPEC-1133)).
3. The test cases themselves specify `readConcern` under the `arguments` field of the "aggregate" operation. AFAIK, none of our spec tests specify per-operation read/write concern options, since that is not a required feature in any of the tests. Instead, there is syntax in some spec test runners for providing such options at the client, database, or collection level.

To address point 2, I would propose that we add this ticket to the [SPEC-1144](https://jira.mongodb.org/browse/SPEC-1144) epic, or at least have its tests depend on [SPEC-1133](https://jira.mongodb.org/browse/SPEC-1133).

To address point 3, I would propose we utilize `operations.collectionOptions` as is done in the transactions spec test runner. AFAIK, the CRUD spec tests also don't use that syntax yet, but it makes sense to introduce that as we standardize on a common format.

Reply:
@jmikola I didn't know about SPEC-1133, but it seems like a great idea to finish that work before adding tests for this. I decided to write these tests the way I because CRUD tests didn't have command monitoring support, and write/read concern tests didn't have support for specifying commands - which resulted in the current hodgepodge.

Reply:
See: https://github.com/mongodb/specifications/pull/453